includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -100,6 +100,8 @@ public function __construct(
 		add_action( "wc_ajax_wcpay_create_payment_intent_$this->stripe_id", [ $this, 'create_payment_intent_ajax' ] );
 		add_action( "wc_ajax_wcpay_update_payment_inten_ $this->stripe_id", [ $this, 'update_payment_intent_ajax' ] );
 		add_action( "wc_ajax_wcpay_init_setup_intent_$this->stripe_id", [ $this, 'init_setup_intent_ajax' ] );
+		add_action( 'woocommerce_after_account_payment_methods', [ $this, 'remove_upe_setup_intent_from_session' ], 10, 0 );
+		add_action( 'woocommerce_order_payment_status_changed', [ __CLASS__, 'remove_upe_payment_intent_from_session' ], 10, 0 );
```

Comment:
Nice catch here, ensuring that this hook fires for every payment method! I don't think it's necessarily related to the changes here, but do you think we should move the `woocommerce_subscription_payment_method_updated` hook up in the main block as well, for the same reason?